secureboot_policy.go

```diff
@@ -830,21 +789,16 @@ func (b *secureBootPolicyGenBranch) processPreOSEvents(events []*tcglog.Event, i
 			b.extendFirmwareVerificationMeasurement(tpm2.Digest(e.Digests[tcglog.AlgorithmId(b.gen.pcrAlgorithm)]))
 		case e.PCRIndex == secureBootPCR:
 			b.extendMeasurement(tpm2.Digest(e.Digests[tcglog.AlgorithmId(b.gen.pcrAlgorithm)]))
+			if e.EventType == tcglog.EventTypeSeparator {
+				seenSecureBootPCRSeparator = true
+			}
 		}
-	}
-
-	if len(events) == 0 {
-		return nil
-	}
 
-	if !initialOSVerificationEvent.measuredInPreOS {
-		return nil
+		if osPresent && seenSecureBootPCRSeparator {
```

Comment:
I'm sure I am missing something but reading the comment:

// Processing of the list of events stops when transitioning from pre-OS to OS-present (ie, a EV_SEPARATOR event is encountered in a
// PCR < 7).

I would have expected  this to be just the osPresent bit or an || ?   if this is correct as is, then the comment above need some expanding/clarification.

Reply:
I've updated this to try to make it clearer.

Reply:
thanks for the updated comment